internal/xds/translator/route.go

```diff
@@ -57,20 +57,25 @@ func buildXdsRoute(httpRoute *ir.HTTPRoute) (*routev3.Route, error) {
 
 		router.Action = &routev3.Route_Route{Route: routeAction}
 	default:
+		routeAction := buildXdsRouteAction(httpRoute.Destination.Name)
+		if httpRoute.Mirrors != nil {
+			routeAction.RequestMirrorPolicies = buildXdsRequestMirrorPolicies(httpRoute.Mirrors)
+		}
 		if httpRoute.BackendWeights.Invalid != 0 {
 			// If there are invalid backends then a weighted cluster is required for the route
-			routeAction := buildXdsWeightedRouteAction(httpRoute)
-			if httpRoute.Mirrors != nil {
-				routeAction.RequestMirrorPolicies = buildXdsRequestMirrorPolicies(httpRoute.Mirrors)
-			}
-			router.Action = &routev3.Route_Route{Route: routeAction}
-		} else {
-			routeAction := buildXdsRouteAction(httpRoute.Destination.Name)
+			routeAction = buildXdsWeightedRouteAction(httpRoute)
 			if httpRoute.Mirrors != nil {
 				routeAction.RequestMirrorPolicies = buildXdsRequestMirrorPolicies(httpRoute.Mirrors)
 			}
-			router.Action = &routev3.Route_Route{Route: routeAction}
 		}
+		// Allow websocket upgrades for HTTP 1.1
+		// Reference: https://developer.mozilla.org/en-US/docs/Web/HTTP/Protocol_upgrade_mechanism
+		routeAction.UpgradeConfigs = []*routev3.RouteAction_UpgradeConfig{
```

Comment:
I do not understand why it is modifying grpc routes as this the only place where websocket configuration is modified. This function should only handle httpRoutes?